instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -145,4 +163,57 @@ private static String stripSensitiveData(@Nullable String url) {
     }
     return url.substring(0, schemeEndIndex + 3) + "REDACTED:REDACTED" + url.substring(atIndex);
   }
+
+  private static String redactQueryParameters(String url) {
+
+    int questionMarkIndex = url.indexOf('?');
+    if (questionMarkIndex == -1 || !containsParamToRedact(url)) {
+      return url;
+    }
+
+    StringBuilder redactedParameters = new StringBuilder();
+    boolean paramToRedact = false; // To be able to skip the characters of the parameters to redact
+    boolean paramNameDetected = false;
+    boolean reference = false;
```

Comment:
these names would make it easier to read for me at least:
```suggestion
    boolean inRedactedParamValue = false; // To be able to skip the characters of the parameters to redact
    boolean inParamValue = false;
    boolean inReference = false;
```